### PR TITLE
improving integrations tests for APIDAO, SubscriptionDAO, FunctionDAO

### PR DESCRIPTION
**Pull request by malinthaprasan**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiResourceDAO.java [hunk 1]
```diff
@@ -172,23 +171,22 @@ static InputStream getBinaryValueForCategory(Connection connection, String apiID
...
         final String query = "SELECT RESOURCE_BINARY_VALUE FROM AM_API_RESOURCES WHERE UUID = ?";
 
         try (PreparedStatement statement = connection.prepareStatement(query)) {
             statement.setString(1, resourceID);
             statement.execute();
 
-            try (ResultSet rs =  statement.getResultSet()) {
+            try (ResultSet rs = statement.getResultSet()) {
                 if (rs.next()) {
-                    Blob blob = rs.getBlob("RESOURCE_BINARY_VALUE");
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiResourceDAO.java]: rs.getBlob() fails in postgres DB

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/APISubscriptionDAOImpl.java [hunk 2]
```diff
@@ -583,53 +582,6 @@ public void updateSubscriptionPolicy(String subId, String policy) throws APIMgtD
...
-    /**
-     * Validates a subscription
-     *
-     * @param apiContext Context of the API
-     * @param apiVersion Version of the API
-     * @param clientId   Client id of the application
-     * @return Subscription Validation Information
-     * @throws APIMgtDAOException If failed to get subscription validation results.
-     */
-    public SubscriptionValidationResult validateSubscription(String apiContext, String apiVersion, String clientId)
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/APISubscriptionDAOImpl.java]: No usages of this method and the SQL query does not seem to be running.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/APISubscriptionDAO.java [hunk 3]
```diff
@@ -233,18 +232,6 @@ void updateSubscriptionStatus(String subId, APIMgtConstants.SubscriptionStatus s
...
     /**
-     * Validates a subscription
-     *
-     * @param apiContext Context of the API
-     * @param apiVersion Version of the API
-     * @param consumerKey Consumer key of the application
-     * @return Subscription Validation Information
-     * @throws APIMgtDAOException   If failed to validat subscription.
-     */
-    SubscriptionValidationResult validateSubscription(String apiContext, String apiVersion, String consumerKey)
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/APISubscriptionDAO.java]: No usages of this method

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/resources/testngDAO.xml [hunk 4]
```diff
@@ -10,6 +10,7 @@
             <class name="org.wso2.carbon.apimgt.core.dao.impl.LabelDAOImplIT" />
             <class name="org.wso2.carbon.apimgt.core.dao.impl.WorkflowDAOIT" />
             <class name="org.wso2.carbon.apimgt.core.dao.impl.PolicyDAOImplIT" />
+            <class name="org.wso2.carbon.apimgt.core.dao.impl.FunctionDAOImplIT" />
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/resources/testngDAO.xml]: why these tests have IT suffix? rather having TestCase?

Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/resources/testngDAO.xml]: Because these are Integration Tests?

Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/resources/testngDAO.xml]: These are not integrations tests right. Are we testing a integration sscenario here?

Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/resources/testngDAO.xml]: We are testing the SQL queries in each DAO with different types of DBs. So can't we consider them as Integration Tests?

Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/resources/testngDAO.xml]: Mhh I don't think so. Then every test case which involves in dao access become a integration test? having word TestCase will be more convinient. Even our integration tests cases names anded with TestCase

**Pull request merged**